### PR TITLE
feat: add support for auth using custom GHES domain

### PR DESCRIPTION
**Pull request by koalaty-code**

### CODE REVIEW

**Review by travi — CHANGES_REQUESTED** [review]
this is great. my one request would be to also include an integration test. 

would you mind adding a scenario to https://github.com/travi/octokit-auth-netrc/blob/master/test/integration/features/netrc.feature? i didnt think through what would be need very deeply, so if a separate feature file is worthwhile, feel free

File: test/integration/features/step_definitions/github-api-steps.js [hunk 1]
```diff
@@ -10,18 +10,20 @@ const user = any.simpleObject();
 Before(function () {
   nock.disableNetConnect();
 
-  githubScope = nock('https://api.github.com/');
+  // githubScope = nock('https://api.github.com/');
```
Comment (travi) [inline test/integration/features/step_definitions/github-api-steps.js:13]: can this be cleaned up?

File: test/integration/features/step_definitions/github-api-steps.js [hunk 2]
```diff
@@ -10,18 +10,20 @@ const user = any.simpleObject();
...
 After(() => {
   nock.enableNetConnect();
   nock.cleanAll();
 
-  assert.isTrue(githubScope.isDone());
+  assert.isTrue(githubScope?.isDone() ?? true);
 });
 
-Given('a personal access token is defined for api.github.com', async function () {
+Given('a personal access token is defined for {}', async function (domain) {
```
Comment (travi) [inline test/integration/features/step_definitions/github-api-steps.js:23]:
```suggestion
Given('a personal access token is defined for {string}', async function (domain) {
```

**Review by travi — CHANGES_REQUESTED** [review]
just a couple of minor details and then i think this is ready to go

File: test/integration/features/step_definitions/github-api-steps.js (hunk 2 above)
Comment (koalaty-code) [reply test/integration/features/step_definitions/github-api-steps.js:23]: `{string}` doesn't work. [Defined](https://github.com/cucumber/cucumber-expressions#parameter-types) as:
> Matches single-quoted or double-quoted strings, for example "banana split" or 'banana split' (but not banana split). Only the text between the quotes will be extracted. The quotes themselves are discarded. Empty pairs of quotes are valid and will be matched and passed to step code as empty strings.

While `{}` is a regex to match anything (`/.*/`).

Comment (travi) [reply test/integration/features/step_definitions/github-api-steps.js:23]: interesting. i guess i never looked deep enough to understand that distinction. i think i normally just go ahead and add the quotes to the feature file to make the typed version work. have thoughts against quoting the values in the feature file. i dont think i feel too strongly either way

File: test/integration/features/step_definitions/github-api-steps.js (hunk 1 above)
Comment (koalaty-code) [reply test/integration/features/step_definitions/github-api-steps.js]: I suppose 😃

File: test/integration/features/step_definitions/github-api-steps.js [hunk 3]
```diff
@@ -9,19 +9,19 @@ const user = any.simpleObject();
...
-
-  githubScope = nock('https://api.github.com/');
 });
 
 After(() => {
   nock.enableNetConnect();
   nock.cleanAll();
 
-  assert.isTrue(githubScope.isDone());
+  assert.isTrue(githubScope?.isDone() ?? true);
```
Comment (travi) [inline test/integration/features/step_definitions/github-api-steps.js:18]: what is the scenario where `githubScope` would not be defined at this point? is there a way we could make it always defined?

File: .nvmrc [hunk 4]
```diff
@@ -1 +1 @@
-16
\ No newline at end of file
+18
```
Comment (travi) [inline .nvmrc:1]: lets make sure we add 18 to the matrix in CI along with this change

**Comment by travi** [comment]
also, with the move to node v18 for the dev version, this no longer needs to use the reusable release workflow for semantic-release v19. could you remove the `-semantic-release-19` from the reusable workflow name?

**Review by travi — APPROVED** [review]

**Pull request merged**